core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,33 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
-          // TODO(bhenning): Consolidate the backend rights domain objects and
-          // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
+          // Check if the response from the backend does not contradict
+          // putParams.
+          var isPrivateResponse = (response.data.status === 'private');
+          if (isPrivateResponse === isPublic) {
+            if (isPublic) {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'published, but response.data.is_private returned true.');
+            } else {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'unpublished, but response.data.is_private returned false.');
+            }
+            if (errorCallback) {
+              errorCallback(response.data);
+            }
+          } else {
+            // An additional fetch is required because the GET response object
```

Comment:
The dictionary that is returned by the PUT response and the GET response are different. This becomes a problem because there's a cache which is initially populated with the GET response (the the subsequent logic that depends on the "schema" being stable such as `isCollectionPrivate()`). Updating the collection with the PUT response results to a different "schema" (`collection_editor.CollectionRightsHandler.get` dictionary vs `rights_manager.get_collection_rights(collection_id).to_dict()` which returns items in `ActivityRights' instance variables.)